src/components/alerts/WalletAlert.tsx

```diff
@@ -7,50 +11,92 @@ import {
   AlertDialogOverlay,
   Box,
   Button,
+  Flex,
   Input,
   InputGroup,
   InputRightElement,
+  Text,
   useToast,
 } from "@chakra-ui/react";
 import { useSession } from "next-auth/react";
-import { useRef, useState } from "react";
+import { ChangeEvent, useRef, useState } from "react";
 import "react-datepicker/dist/react-datepicker.css";
 
 type Props = {
   isOpen: boolean;
   onCancel: () => void;
   refetch: () => void;
+  balance: number;
+};
+export type LightningResponse = {
+  status: string;
+  tag: string;
+  commentAllowed: number;
+  callback: string;
+  metadata: string;
+  minSendable: number;
+  maxSendable: number;
+  payerData: PayerData;
+  nostrPubkey: string;
+  allowsNostr: boolean;
+};
+
+export type PayerData = {
+  name: Email;
+  email: Email;
+  pubkey: Email;
+};
+
+export type Email = {
+  mandatory: boolean;
 };
 
-const WalletAlert = ({ isOpen, onCancel, refetch }: Props) => {
+const WalletAlert = ({ isOpen, onCancel, refetch, balance }: Props) => {
   const cancelRef = useRef<HTMLButtonElement>(null);
 
   const { data: sessionData } = useSession();
   const sessionDataId = sessionData?.user?.id;
   const toast = useToast();
   const [invoiceInput, setInvoiceInput] = useState("");
+  const [amountToSend, setAmountToSend] = useState<number | string>();
+  const [step, setStep] = useState(1);
+  const [lightningData, setLightningData] = useState<
+    LightningResponse | undefined
```

Comment:
When you omit a default value and pass a type `T`, it infers as `T | undefined` so no need for the undefined.

Reply:
Yh